File: tareas_go/commands/cambiar_estado.go
package commands

import (
	"fmt"
	"tareas_go/estructuras"
)

func (g *GestorTareas) CambiarEstado(id int, nuevo estructuras.Estado) error {
	for i := range g.tareas {
		if g.tareas[i].ID == id {
			g.tareas[i].Estado = nuevo
			g.GuardarTareasJSON()
			fmt.Printf("Estado de la tarea %d cambiado a %s.\n", id, estructuras.EstadoToString(nuevo))
			return nil
		}
	}
	return fmt.Errorf("tarea no encontrada")
}

File: tareas_rust/src/commands/cambiar_estado.rs
use crate::commands::persistencia;
use crate::commands::gestor::GestorTareas;
use crate::estructuras::Estado;

pub fn cambiar_estado(g: &mut GestorTareas, id: i32, nuevo: Estado) -> Result<(), String> {
    if let Some(t) = g.tareas.iter_mut().find(|t| t.id == id) {
        t.cambiar_estado(nuevo);
        persistencia::guardar_tareas_json(&g.tareas)?;
        println!("Estado de la tarea {} cambiado correctamente.", id);
        return Ok(());
    }
    Err(format!("Tarea {} no encontrada.", id))
}
